test/utils/utils.go

```diff
@@ -456,3 +458,72 @@ func PortForwardService(ctx context.Context, namespace, serviceName string, port
 	// Port forward to the resolved pod using the target port
 	return PortForwardPod(ctx, namespace, pod.Name, targetPort)
 }
+
+// MakeGatewayRequest establishes a port-forward to the service, makes an HTTP request, and cleans up.
+// The operation is wrapped in Eventually to handle transient network errors and pod restarts.
+// Non-2xx HTTP status codes are returned successfully (not treated as errors), allowing callers
+// to verify specific status codes like 404.
+func MakeGatewayRequest(
+	namespace, serviceName string,
+	servicePort int,
+	endpoint, method string,
+	payload interface{},
+	timeout, interval time.Duration,
+) (body []byte, statusCode int, err error) {
+	// Variables to capture results from the Eventually closure
+	var capturedBody []byte
+	var capturedStatusCode int
+
+	// Eventually will retry until the function returns nil (success) or timeout occurs
+	Eventually(func() error {
```

Comment:
Keep Eventually, Except, etc. in the test code to avoid mixing it. It should be up to the concrete test to decide if an Eventually is needed and how long to wait.